parquet_to_line_protocol/src/lib.rs

```diff
@@ -0,0 +1,237 @@
+use schema::Schema;
+
+use snafu::{OptionExt, ResultExt, Snafu};
+
+mod batch;
+
+use batch::convert_to_lines;
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("Invalid path: {:?}: {}", path, source))]
+    Path {
+        path: PathBuf,
+        source: object_store::path::Error,
+    },
+
+    #[snafu(display("Error listing: {:?}: {}", object_store_path, source))]
+    ObjectStorePath {
+        object_store_path: ObjectStorePath,
+        source: object_store::Error,
+    },
+
+    #[snafu(display(
+        "Can not find IOx metadata in parquet metadata. Could not find {}",
+        METADATA_KEY
+    ))]
+    MissingMetadata {},
+
+    #[snafu(display("Error reading IOx metadata: {}", source))]
+    Metadata {
+        source: parquet_file::metadata::Error,
+    },
+
+    #[snafu(display("Error infering IOx schema: {}", source))]
+    InferringSchema {
+        source: datafusion::error::DataFusionError,
+    },
+
+    #[snafu(display("Error reading IOx schema: {}", source))]
+    Schema { source: schema::Error },
+
+    #[snafu(display("Error in processing task: {}", source))]
+    Task { source: tokio::task::JoinError },
+
+    #[snafu(display("Error converting: {}", message))]
+    Conversion { message: String },
+
+    #[snafu(display("Error executing: {}", source))]
+    ExecutingStream {
+        source: datafusion::error::DataFusionError,
+    },
+
+    #[snafu(display("IO Error: {}", source))]
+    IO { source: std::io::Error },
+}
+
+/// Converts a parquet file that was written by IOx from the local
+/// file system path specified to line protocol and writes those bytes
+/// to `output`, returning the writer on success
+pub async fn convert_file<W, P>(path: P, mut output: W) -> Result<W, Error>
+where
+    P: AsRef<Path>,
+    W: Write,
+{
+    let path = path.as_ref();
+    let object_store_path =
+        ObjectStorePath::from_filesystem_path(path).context(PathSnafu { path })?;
+
+    // Fire up a parquet reader, read the batches, and then convert
+    // them asynchronously in parallel
+
+    let object_store = Arc::new(LocalFileSystem::new()) as Arc<dyn ObjectStore>;
+    let object_store_url = ObjectStoreUrl::local_filesystem();
+
+    let object_meta = object_store
+        .head(&object_store_path)
+        .await
+        .context(ObjectStorePathSnafu { object_store_path })?;
+
+    let reader = ParquetFileReader::try_new(object_store, object_store_url, object_meta).await?;
+
+    // Determines the measurement name from the IOx metadata
+    let schema = reader.schema();
+    let encoded_meta = schema
+        .metadata
+        .get(METADATA_KEY)
+        .context(MissingMetadataSnafu)?;
+
+    let iox_meta = IoxMetadata::from_base64(encoded_meta.as_bytes()).context(MetadataSnafu)?;
+
+    // Attempt to extract the IOx schema from the schema stored in the
+    // parquet file. This schema is where information such as what
+    // columns are tags and field sfields is stored
```

Comment:
Extra fields snuck in here:

```suggestion
    // columns are tags and fields is stored
```